lib/ansible/modules/async_wrapper.py

```diff
@@ -231,14 +232,23 @@ def main():
     jobdir = os.path.expanduser(async_dir)
     job_path = os.path.join(jobdir, jid)
 
-    if not os.path.exists(jobdir):
-        try:
-            os.makedirs(jobdir)
-        except Exception:
-            print(json.dumps({
-                "failed": 1,
-                "msg": "could not create: %s" % jobdir
-            }))
+    err = None
+    try:
+        os.makedirs(jobdir)
+    except OSError as e:
+        if e.errno != errno.EEXIST:
+            err = (e, traceback.format_exc())
+    except Exception as e:
+        err = (e, traceback.format_exc())
```

Comment:
Readability-wise, it'd be cleaner to put this into a helper function.

Reply:
Done